src/main/java/seedu/meetingjio/timetables/MasterTimetable.java

```diff
@@ -77,6 +88,76 @@ public int getSize() {
         return timetables.size();
     }
 
+    public ArrayList<Meeting> getMeetingList() {
+        return meetingList;
+    }
+
+    public boolean checkIfClash(Meeting meeting) {
+        for (Timetable timetable : timetables) {
+            if (checkMeetingOverlap(timetable, meeting)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean checkMeetingOverlap(Timetable timetable, Meeting meeting) {
+        for (Event event : timetable.getList()) {
+            if (meeting.overlaps(event)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean checkIfMeetingExistsAlready(Meeting meeting) {
+        for (Event event : meetingList) {
+            if (meeting.equals(event)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public String addMeetingToEveryoneTimetable(Meeting meeting) {
```

Comment:
Good to extract addMeeting method to the MasterTimetable class instead of doing it in prepareAddMeeting function!